ibm/resource_ibm_is_network_acl_rule_test.go

```diff
@@ -0,0 +1,118 @@
+// Copyright IBM Corp. 2017, 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package ibm
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestNetworkACLRule_basic(t *testing.T) {
+	var nwACLRule string
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: checkNetworkACLRuleDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckIBMISNetworkACLRuleConfig1(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIBMISNetworkACLRuleExists("ibm_is_network_acl_rule.testacc_nacl", nwACLRule),
```

Comment:
Please add testcases for multiple scenarios 
creating icmp rule
creating tcp rule
creating UDP rule
creating all protocol rule
and the testcases should cover the update scenarios also